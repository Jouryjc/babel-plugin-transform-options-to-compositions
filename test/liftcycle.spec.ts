import plugin from '../lib/index'
import { transformSync } from '@babel/core'

describe('lifecycle', () => {
  test('lifecycle', () => {
    const { code } = transformSync(
      `
      export default {
        beforeUnmount () {
          console.log('before unmount!')
        }
      }
    `,
      {
        presets: [
          [
            '@babel/preset-env',
            {
              modules: false,
            },
          ],
        ],
        plugins: [plugin],
      }
    )

    expect(code).toMatchSnapshot('beforeUnmount')
  })
})
