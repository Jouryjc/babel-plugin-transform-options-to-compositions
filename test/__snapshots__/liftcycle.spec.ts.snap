// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lifecycle lifecycle: beforeUnmount 1`] = `
"import { onBeforeUnmount } from 'vue';
onBeforeUnmount(() => {
  console.log('before unmount!');
});"
`;
